docs/modules/ROOT/pages/references/architecture/espejote-in-cluster-templating-controller.adoc

```diff
@@ -0,0 +1,502 @@
+The re-renders are limited by the specific unique resource.
+<10> Context is a list of resources that are available in the Jsonnet template.
+They're not watched.
+<11> `def` defines a variable in the Jsonnet template.
+<12> `resource` is a Kubernetes resource that's available in the Jsonnet template.
+<13> `cache` is optional and can be used to setup a controller-runtime cache for the resource.
+The default is true.
+<14> `std.extVar` is used to access the trigger data.
+The full manifest is available if using the `watchResource` trigger.
+<15> `std.extVar` is used to access the context.
+It returns a object with all defined resources as keys.
+<16> The template can return a list of resources or a single resource.
+<17> The namespace of the resource will always be overwritten to the namespace of the `ManagedResource`.
+<18> Resource variables are always a list with zero or more items.
+
+
+=== `ClusterManagedResource`
+
+[source,yaml]
+----
+apiVersion: espejote.io/v1alpha1
+kind: ClusterManagedResource
+metadata:
+  name: default-network-policies
+spec:
+  triggers:
+  - watchResource:
+      apiVersion: v1
+      kind: Namespace <1>
+      labelSelector:
+        matchExpressions:
+        - key: inject-cm.syn.tools
+          operator: Exists
+  template: |
+    local cm = std.extVar('trigger').resource; <2>
+
+    [
+      {
+        metadata: {
+          name: 'injected-cm' + c.metadata.name,
+          namespace: c.metadata.namespace, <3>
+        }
+        stringData: {
+          'my-key': 'my-value',
+        }
+      }
+    ]
+----
+<1> Cluster scoped `ClusterManagedResource` can access all resources in the cluster.
+This includes cluster scoped resources.
+<2> The full manifest of the triggering is available.
+<3> A cluster managed resource can create resources in any namespace and thus needs to set the namespace.
+
+== Sample use-cases
+
+=== Sync upgrade notifications with a ArgoCD sync hook and bash script
+
+We want to show a notification in the OpenShift Console when a minor upgrade is available.
+To implement this we added a job triggered by an ArgoCD sync.
+The job is reading another resource and creating a console notification manifest from it.
+
+* https://github.com/appuio/component-openshift4-console/blob/740628ebf3822ea82a64fade1e42eb9ff52f67c7/tests/golden/upgrade-notification/openshift4-console/openshift4-console/31_upgrade_notification.yaml#L63[component-openshift4-console ArgoCD sync hook]
+* https://github.com/appuio/component-openshift4-console/blob/740628ebf3822ea82a64fade1e42eb9ff52f67c7/component/scripts/create-console-notification.sh[component-openshift4-console bash script]
+
+=== Sync OpenShift Console TLS Secret with a bash "reconciler"
+
+Cert-manager only allows to create certificates in the same namespace as the ingress.
+OpenShift Console ingress is in the `openshift-console` namespace, but the secret needs to be applied in the `openshift-config` namespace.
```

Comment:
This isn't directly coupled with the OpenShift console ~~ingress~~ route

```suggestion
The OpenShift Console route is in the `openshift-console` namespace, but the secret needs to be applied in the `openshift-config` namespace.
However, we must create the `Certificate` resource in `openshift-console`, since otherwise the OpenShift ingress doesn't admit the HTTP challenge ingress.
```